frontend/src/main/scala/bloop/exec/ProcessConfig.scala

```diff
@@ -0,0 +1,97 @@
+package bloop.exec
+
+import java.io.File.{separator, pathSeparator}
+import java.net.URLClassLoader
+
+import scala.util.control.NonFatal
+
+import bloop.io.AbsolutePath
+import bloop.logging.{Logger, ProcessLogger}
+
+/**
+ * Configures how to start new processes to run Java code.
+ */
+sealed trait ProcessConfig {
+
+  /**
+   * The full classpath with which the code should be executed
+   */
+  def classpath: Array[AbsolutePath]
+
+  /**
+   * Run the main function in class `className`, passing it `args`.
+   *
+   * @param className The fully qualified name of the class to run.
+   * @param args      The arguments to pass to the main method.
+   * @param logger    Where to log the messages from execution.
+   * @return 0 if the execution exited successfully, a non-zero number otherwise.
+   */
+  def runMain(className: String, args: Array[String], logger: Logger): Int
+}
+
+object ProcessConfig {
+  def apply(javaEnv: JavaEnv, classpath: Array[AbsolutePath]): ProcessConfig =
+    if (javaEnv.fork) Fork(classpath, javaEnv)
+    else InProcess(classpath)
+
+  /** The code returned after a successful execution. */
+  final val EXIT_OK = 0
+
+  /** The code returned after the execution errored. */
+  final val EXIT_ERROR = 1
+}
+
+/**
+ * Configuration to execute Java code without starting a new JVM.
+ */
+case class InProcess(classpath: Array[AbsolutePath]) extends ProcessConfig {
+
+  override def runMain(className: String, args: Array[String], logger: Logger): Int = {
+    logger.debug(s"Running '$className' in process.")
+    logger.debug(s"  Classpath: ${classpath.map(_.syntax).mkString(pathSeparator)}")
+    val entries = classpath.map(_.toFile.toURI.toURL)
```

Comment:
Can we cache new classloaders that we create? Or maybe you can leave this to me, I think we're not caching the test classloader either. I can do it in another PR.

Reply:
Caching comes as part of #136 😄